src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -184,6 +188,7 @@ private void switchTimelines(final Timeline activeTimeline, final Timeline nextT
                     final Timeline.StoragePosition storagePosition =
                             topicRepositoryHolder.createStoragePosition(activeTimeline);
                     activeTimeline.setLatestPosition(storagePosition);
+                    scheduleTimelineCleanup(activeTimeline);
```

Comment:
It will be closer to the date, when topic was really created, so the clean up time will be more precise, honestly, so much time passed since I wrote it I do not even remember why :)